src/main/scala/is/hail/distributedmatrix/GridPartitioner.scala

```diff
@@ -10,7 +10,16 @@ case class GridPartitioner(blockSize: Int, rows: Long, cols: Long, transposed: B
 
   val rowPartitions: Int = ((rows - 1) / blockSize + 1).toInt
   val colPartitions: Int = ((cols - 1) / blockSize + 1).toInt
-
+  
+  val truncatedBlockRow: Int = (rows / blockSize).toInt
```

Comment:
This is a slightly unrelated comment prompted by the `isTranspose` stuff below:

I don't think GridPartitioner should carry a `transposed` variable.  `transpose` should transpose each element (essentially free since Breeze tracks the transposes) and then reorders the partition (i, j) <=> (j, i) (also essentially free).  Then we don't ever need to keep track of transposition which is error prone.

In fact, I'm slightly skeptical of `rowPartitionRows(i: Int)` since `i` should really be `j` if `transpose = true`.  I think you should fix this.